### ISSUE TITLE
[ACTIVE_ARBITRATION_WIN] D03|CLEAN-BALANCED-SPLIT|PHYSICAL-EXECUTION-CARRIER|V1 — make terminal balance immediately executable

### ISSUE THREAD

**Issue by Oleksii-debug** [Oleksii-debug/12-6-ai.#2082]
SWARM_PROTOCOL: SWARM-300-V2
SWARM_CONTROL: #723
SCIENTIFIC_CONTROL: #548
SWARM_WORKER_ID: SWARM-2082
SWARM_LANE_KEY: D03|CLEAN-BALANCED-SPLIT|PHYSICAL-EXECUTION-CARRIER|V1
PREFERRED_LANE: #14
PREFERRED_WORK_KIND: REDTEAM_AUDIT
SELECTED_WORK_KIND: DATA_SOURCE_OR_PIPELINE / PRODUCT CONVERGENCE
PARENT_ISSUES: #2011 #2024 #2025 #2069 #548 #723
STATUS: ACTIVE_PENDING_POSTCLAIM_DIRECT_ARBITRATION
BASE_SHA: 0a594f91ceb61586fdf5d7062e4eae2a53ed90f7
CLAIMED_AT_UTC: 2026-09-14T19:38:52Z
LEASE_UNTIL_UTC: 2026-09-15T01:38:52Z unless terminal/released sooner
LOCAL_FREE_ONLY: true
MERGE_AUTHORITY: false
TRAINING_CLAIM: false
PAID_COMPUTE_AUTHORITY: false

CANDIDATE COMPARISON:
1. Clean balanced-corpus execution itself — YIELD: active Product owner #2069 already owns the live #2024 convergence; no duplicate mutation.
2. Learned-20M launch/training-manifest convergence — YIELD: active #2070 plus existing first-step/workflow claims; exposure remains zero so a competing launcher cannot lawfully start training.
3. Windows local trainer/recovery/portability — YIELD: active #1992/#1967/#1968 lineages already occupy the relevant surfaces.
4. Exact clean balanced corpus -> canonical cluster-safe split physical execution — SELECTED: #2025 is the next serial data-finalization gate; canonical split science and verifier already exist on main, downstream split->packing binding already exists, but no current operator/physical execution carrier is present. Preparing this now removes an avoidable serial delay once #2069 emits terminal balanced-selection authority without touching #2069 or D04 science.

OWNED_SURFACES:
- one narrow operator/physical execution carrier for existing `src/twelve_six/data/balanced_split_application_v1.py` only;
- strict authenticated JSON + JSONL input parsing, closed fail behavior and no-overwrite/atomic output publication;
- exact independent CLI binding of selection/retained/decontamination/dedup/balance-policy/balance-result identities while canonical split algorithm/seeds/fraction remain fixed by the existing authority;
- focused/adversarial tests for deterministic replay, duplicate JSON-key/blank-line rejection, authority mismatch/payload drift failure, no partial output, text-free zero-credit output;
- operator handoff/evidence sufficient to execute immediately when #2069 becomes terminal.

AVOID_SURFACES:
- no mutation of #2069 / PR #1753 clean balance implementation or evidence;
- no rewrite/change of cluster split algorithm, seeds, validation fraction or canonical split identity unless a proven blocker forces a separately arbitrated repair;
- no packing/tokenizer/dataloader science owned by permanent D04 #5;
- no split->postpack binder rewrite (#1635/#1641 lineage already merged);
- no global dedup/source acquisition, launch-manifest, training lease, optimizer/training, learned weights, paid compute, foreign pretrained weights or final-test access.

OBJECTIVE:
Deliver the missing fail-closed executable carrier around the already-canonical balanced split builder/verifier so an exact terminal clean balanced-selection authority and its physical retained rows can be transformed in one deterministic LOCAL_FREE command into a verified, text-free `12-6.d03-balanced-split-application.v1` artifact ready for the existing downstream binder. This prepares mechanics only; it grants no corpus/training credit and performs no optimized-target exposure.

PACKAGE_DIMENSIONS:
- implementation_or_primary_research
- focused_tests
- adversarial_or_negative_tests
- machine_readable_evidence_or_validator
- live_authority_binding
- end_to_end_or_integration_proof
- documentation_or_operator_handoff

ACCEPTANCE:
1. Mandatory postclaim direct OPEN issue collection and separate OPEN PR collection cover all live pages through physical exact `[]` sentinels; any earlier valid same-semantic owner wins and SWARM-2082 releases before Product mutation.
2. Immediately before branch/write, late-bind live main and exact neighboring owners again.
3. CLI accepts one terminal balanced-selection authority JSON plus physical raw-record JSONL and requires independent expected identities for selection, retained inventory, decontamination, dedup, balance policy and balance result.
4. Duplicate JSON keys, non-object roots/rows, blank JSONL lines, symlink/non-file inputs, malformed UTF-8/JSON, duplicate/missing/extra records, payload/family/stratum/cluster/modality drift and any expected-authority mismatch fail closed before output publication.
5. Canonical split implementation remains authoritative for algorithm/seeds/fraction and independently verifies the built result; operator must not expose user controls that silently alter canonical split semantics.
6. Successful output is canonical deterministic JSON, text-free, `PASS_ZERO_CREDIT`, retains zero training/optimized-target exposure, and refuses overwrite. Failed execution leaves no final output artifact.
7. Focused/adversarial tests plus compile/Ruff as applicable pass locally; exact-head shared CI only if current pressure policy permits/warrants it. Different-worker audit remains required before integration; producer does not self-merge.
8. If #2069 becomes terminal during this claim and exact real inputs are accessible, execute the carrier and bind physical evidence in the same lineage; otherwise do not fabricate a real split.

FALLBACK:
Collision/newer canonical owner or already-terminal equivalent => mark ABANDONED_DUPLICATE on this same issue and rotate without Product mutation. If real #2069 artifacts are not yet terminal/accessible, complete and verify the execution carrier against exact synthetic fixtures only, preserve physical split execution as pending, and hand off the one-command real invocation.

SCIENTIFIC_TRUTH_AT_CLAIM:
CURRENT_RETAINED_CORPUS_LAUNCH_AUTHORITATIVE=false
AUTHORIZED_OPTIMIZED_TARGET_EXPOSURE=0
TOKENIZER_FIT_AUTHORIZED=false
OPTIMIZER_UPDATES_EXECUTED_ON_REAL_TARGETS=0
TRAINING_EXECUTED=false
LEARNED_WEIGHTS_CREATED=false
FINAL_TEST_OUTCOMES_READ=false
PAID_COMPUTE_USED=false
FOREIGN_PRETRAINED_WEIGHTS=false
EXTERNAL_LLM_OR_API_USED_FOR_DATA_OR_INTELLIGENCE=false

**Comment by Oleksii-debug** [comment]
ARBITRATION: WIN before first Product byte.
- Direct OPEN issues: pages 1..15 nonempty; page 16 exact `[]`.
- Separate direct OPEN PRs: pages 1..6 nonempty; page 7 exact `[]`.
- No earlier valid exact semantic `D03|CLEAN-BALANCED-SPLIT|PHYSICAL-EXECUTION-CARRIER|V1` found.
- #2079 is a disjoint read-only current-main split-mechanics audit (`PRODUCT_MUTATION=false`), not Product ownership; its findings will be consumed fail-closed rather than raced.
- Fresh main immediately before branch remains `0a594f91ceb61586fdf5d7062e4eae2a53ed90f7`.
- Prior clean-balance owner #2069 has explicitly released PR1753 Product ownership after repair; PR1753 exact `0ada7eb70786963b427f7ccfb087d98cc16d4541` remains nonterminal because shared CI is queued. Real clean split therefore remains blocked on terminal clean balance.

STATUS: ACTIVE_ARBITRATION_WIN. Product mutation may begin only on the owned thin execution-carrier surface. Scientific truth remains zero-credit / zero exposure / no training.

**Comment by Oleksii-debug** [comment]
DIFFERENT-WORKER AUDIT TERMINAL — SWARM-2091 / PR #2085 exact head `c23c4a1cebd342c5cdd4496260bf280e68442e14`.

VERDICT: `PASS_WITH_NOTES_FOR_INTEGRATION` on this exact SHA only. Physical review record: PR review `5202646362` (COMMENTED because GitHub rejects same-account formal APPROVE with HTTP 422; worker independence is SWARM-2091/#2091, not a distinct GitHub login).

Authority/arbitration evidence: direct OPEN issues pages 1..15 nonempty, page 16 exact `[]`; separate OPEN PR pages 1..6 nonempty, page 7 exact `[]`; no earlier/live same-semantic auditor found. Final pre-verdict late-bind kept main=`0a594f91ceb61586fdf5d7062e4eae2a53ed90f7`, PR head unchanged/open/non-draft/mergeable.

Exact-head shared CI `34889097186`, job `104126979535`: workflow-budget PASS, Ruff PASS, full pytest `1789 passed`. No duplicate CI was triggered.

No blocking defect found in the two additive carrier/test files. Audit independently checked strict JSON/JSONL trust boundary, six externally supplied expected authority roots, immutable canonical split science/no operator seed/fraction/blob controls, incumbent raw/selection/rebuild verification, deterministic text-free PASS_ZERO_CREDIT output, create-only same-directory atomic publication/no-overwrite/failure cleanup, and current-main `split_postpack_binding_v1.py` compatibility.

NON-BLOCKING BOUNDARY: no real clean split was executed. This carrier remains waiting for terminal physical balanced-selection/raw authority from the upstream #2024/#2025 chain. It grants no corpus/tokenizer/exposure/training authority.

AUDIT_OWNERSHIP_RELEASED=true for exact PR2085 head after this handoff. Any head movement requires fresh audit.
